core/src/display_object/movie_clip.rs

```diff
@@ -188,6 +188,9 @@ pub struct MovieClipData<'gc> {
 
     /// Attached audio (AVM1)
     attached_audio: Option<NetStream<'gc>>,
+
+    // If this movie was loaded from ImportAssets(2), this will be the parent movie.
+    importer_movie: Option<Arc<SwfMovie>>,
```

Comment:
As far as I can tell, the motivating idea for this field is that when A.swf imports B.swf, and then B.swf exports assets, B.swf's root `MovieClip` can start copying asset definitions into A.swf's library. This does seem like it'd work, though I'm not entirely sure if managing import/export relationships like this is the correct solution.

For example: if later on we add C.swf to our movie network, and C.swf imports B.swf again, the way this is currently implemented, we have to redownload and reexecute a second copy of the movie, creating two copies of every asset that won't compare equal. Flash Player might actually work this way, but if it didn't, then this solution wouldn't work. We'd need to have importer/exporter relationships centralized somewhere (probably in `Library`).

Reply:
Yes you guessed right. Unfortunately those stuff are not in avmplus and a harder to get it right. Having working games is the main priority for me, a future refactoring is definitely not excluded but I would say out of scope for this pr

Reply:
Flash *does* seem to work this way. From my testing, two imports to the same URL will trigger a download twice - which means it's *theoretically* possible to serve a different movie each time and get a different result.

I think we can probably pretend that's not the case and cache it in the future though, somehow